pkg/sitewise/datasource.go

```diff
@@ -30,6 +33,21 @@ type Datasource struct {
 	GetClient         clientGetterFunc
 }
 
+type disableHostPrefixMiddleware struct{}
+
+func (m *disableHostPrefixMiddleware) ID() string {
+	return "DisableHostPrefixMiddleware"
+}
+
+func (m *disableHostPrefixMiddleware) HandleInitialize(
+	ctx context.Context, in middleware.InitializeInput, next middleware.InitializeHandler,
+) (
+	out middleware.InitializeOutput, metadata middleware.Metadata, err error,
+) {
+	ctx = smithyhttp.SetHostnameImmutable(ctx, true)
+	return next.HandleInitialize(ctx, in)
+}
+
```

Comment:
This is to prevent the endpoint from automatically getting `api` prepended to it. It's used below only if the region is set to Edge.

E.g. when you configure the endpoint in the config editor to be something like `https://localhost:8443`, having this middleware prevents the client from making a call to `https://api.localhost:8443` instead.